src/propagate-tor-test_cuda.cu

```diff
@@ -0,0 +1,1048 @@
+    errorProp.data[bsize*PosInMtrx(1,4,6) + it] = k*(sinPorT*dadphi*cosa + cosPorT*dadphi*sina + sinPorT*cosa + cosPorT*sina - sinPorT)*pt;
+    errorProp.data[bsize*PosInMtrx(1,5,6) + it] = 0.f;
+
+    //no trig approx here, theta can be large
+    cosPorT=std::cos(thetain);
+    sinPorT=std::sin(thetain);
+    //redefine sinPorT as 1./sinPorT to reduce the number of temporaries
+    sinPorT = 1.f/sinPorT;
+
+    setz(outPar,it, z(inPar,it) + k*alpha*cosPorT*pt*sinPorT);
+
+    errorProp.data[bsize*PosInMtrx(2,0,6) + it] = k*cosPorT*dadx*pt*sinPorT;
+    errorProp.data[bsize*PosInMtrx(2,1,6) + it] = k*cosPorT*dady*pt*sinPorT;
+    errorProp.data[bsize*PosInMtrx(2,2,6) + it] = 1.f;
+    errorProp.data[bsize*PosInMtrx(2,3,6) + it] = k*cosPorT*(iptin*dadipt-alpha)*pt*pt*sinPorT;
+    errorProp.data[bsize*PosInMtrx(2,4,6) + it] = k*dadphi*cosPorT*pt*sinPorT;
+    errorProp.data[bsize*PosInMtrx(2,5,6) + it] =-k*alpha*pt*sinPorT*sinPorT;
+
+    setipt(outPar,it, iptin);
+
+    errorProp.data[bsize*PosInMtrx(3,0,6) + it] = 0.f;
+    errorProp.data[bsize*PosInMtrx(3,1,6) + it] = 0.f;
+    errorProp.data[bsize*PosInMtrx(3,2,6) + it] = 0.f;
+    errorProp.data[bsize*PosInMtrx(3,3,6) + it] = 1.f;
+    errorProp.data[bsize*PosInMtrx(3,4,6) + it] = 0.f;
+    errorProp.data[bsize*PosInMtrx(3,5,6) + it] = 0.f;
+
+    setphi(outPar,it, phi(inPar,it)+alpha );
+
+    errorProp.data[bsize*PosInMtrx(4,0,6) + it] = dadx;
+    errorProp.data[bsize*PosInMtrx(4,1,6) + it] = dady;
+    errorProp.data[bsize*PosInMtrx(4,2,6) + it] = 0.f;
+    errorProp.data[bsize*PosInMtrx(4,3,6) + it] = dadipt;
+    errorProp.data[bsize*PosInMtrx(4,4,6) + it] = 1.f+dadphi;
+    errorProp.data[bsize*PosInMtrx(4,5,6) + it] = 0.f;
+
+    settheta(outPar,it, thetain);
+
+    errorProp.data[bsize*PosInMtrx(5,0,6) + it] = 0.f;
+    errorProp.data[bsize*PosInMtrx(5,1,6) + it] = 0.f;
+    errorProp.data[bsize*PosInMtrx(5,2,6) + it] = 0.f;
+    errorProp.data[bsize*PosInMtrx(5,3,6) + it] = 0.f;
+    errorProp.data[bsize*PosInMtrx(5,4,6) + it] = 0.f;
+    errorProp.data[bsize*PosInMtrx(5,5,6) + it] = 1.f;
+  }
+
+  MultHelixProp(&errorProp, inErr, &temp);
+  MultHelixPropTransp(&errorProp, &temp, outErr);
+}
+
+inline void transferAsyncTrk(MPTRK* trk_dev, MPTRK* trk, cudaStream_t stream){
+
+  cudaMemcpyAsync(trk_dev, trk, nevts*nb*sizeof(MPTRK), cudaMemcpyHostToDevice, stream);
+  cudaMemcpyAsync(&trk_dev->par, &trk->par, sizeof(MP6F), cudaMemcpyHostToDevice, stream);
+  cudaMemcpyAsync(&((trk_dev->par).data), &((trk->par).data), 6*bsize*sizeof(float), cudaMemcpyHostToDevice, stream);
+  cudaMemcpyAsync(&trk_dev->cov, &trk->cov, sizeof(MP6x6SF), cudaMemcpyHostToDevice, stream);
+  cudaMemcpyAsync(&((trk_dev->cov).data), &((trk->cov).data), 36*bsize*sizeof(float), cudaMemcpyHostToDevice, stream);
+  cudaMemcpyAsync(&trk_dev->q, &trk->q, sizeof(MP1I), cudaMemcpyHostToDevice, stream);
+  cudaMemcpyAsync(&((trk_dev->q).data), &((trk->q).data), 1*bsize*sizeof(int), cudaMemcpyHostToDevice, stream);  
```

Comment:
These transfers are unnecessary, right? (since the first copy transfers them all)